--- backend/pixma_mp150.c.orig	2011-01-15 17:01:28.000000000 -0800
+++ backend/pixma_mp150.c	2011-04-16 16:32:52.797216260 -0700
@@ -160,6 +160,7 @@
 /* PIXMA 2010 vintage */
 #define MX340_PID 0x1741
 #define MX350_PID 0x1742
+#define MP495_PID 0x1747
 #define MX870_PID 0x1743
 
 /* Generation 4 XML messages that encapsulates the Pixma protocol messages */
@@ -1559,16 +1560,16 @@
   mp150_get_status
 };
 
-#define DEVICE(name, model, pid, dpi, w, h, cap) { \
-        name,              /* name */               \
-        model,             /* model */              \
-        CANON_VID, pid,    /* vid pid */            \
-        0,                 /* iface */              \
-        &pixma_mp150_ops,  /* ops */                \
-        dpi, 2*(dpi),      /* xdpi, ydpi */         \
-        w, h,              /* width, height */      \
-        PIXMA_CAP_EASY_RGB|PIXMA_CAP_GRAY|          \
-        PIXMA_CAP_GAMMA_TABLE|PIXMA_CAP_EVENTS|cap  \
+#define DEVICE(name, model, pid, dpi, w, h, cap) {			\
+        name,              /* name as quoted string*/			\
+        model,             /* model, upper case letters */		\
+        CANON_VID, pid,    /* USB Vendor and Product number as integer */\
+        0,                 /* iface */					\
+        &pixma_mp150_ops,  /* virtual table for scanner methods */	\
+        dpi, 2*(dpi),      /* horizontal (X) dots per inch, Y always double */\
+        w, h,              /* max width, height in points (1/72 inch) */\
+        PIXMA_CAP_EASY_RGB|PIXMA_CAP_GRAY|				\
+        PIXMA_CAP_GAMMA_TABLE|PIXMA_CAP_EVENTS|cap			\
 }
 
 #define END_OF_DEVICE_LIST DEVICE(NULL, NULL, 0, 0, 0, 0, 0)
@@ -1649,6 +1650,8 @@
   DEVICE ("Canon PIXMA MP490", "MP490", MP490_PID, 1200, 638, 877, PIXMA_CAP_CIS),
   DEVICE ("Canon PIXMA MP250", "MP250", MP250_PID, 600, 638, 877, PIXMA_CAP_CIS),
   DEVICE ("Canon PIXMA MP270", "MP270", MP270_PID, 1200, 638, 877, PIXMA_CAP_CIS),
+  /* jimc added; product inception about 2010-07-xx replacing the MP480 */
+  DEVICE ("Canon PIXMA MP495", "MP495", MP495_PID, 1200, 638, 877, PIXMA_CAP_CIS),
 
   /* Generation 4 CCD */
   DEVICE ("Canon MP990 series", "MP990", MP990_PID, 4800, 638, 877, PIXMA_CAP_CCD | PIXMA_CAP_TPU),
