--- configure.in.primascan	2011-04-01 10:21:21.000000000 +0200
+++ configure.in	2011-04-01 10:21:39.000000000 +0200
@@ -562,7 +562,7 @@
         pixma plustek plustek_pp qcam ricoh rts8891 s9036 \
         sceptre sharp sm3600 sm3840 snapscan sp15c st400 \
         stv680 tamarack teco1 teco2 teco3 test u12 umax 
-        umax_pp umax1220u v4l xerox_mfp p5"
+        umax_pp umax1220u v4l xerox_mfp p5 primascan"
 
 # If user specifies backends manually then cause configure
 # to fail if its detected it can't be compiled.  If we
--- backend/Makefile.am.primscan	2011-04-01 10:22:04.000000000 +0200
+++ backend/Makefile.am	2011-04-01 10:24:11.000000000 +0200
@@ -191,7 +191,7 @@
     libtamarack.la libtest.la libteco1.la \
     libteco2.la libteco3.la libu12.la libumax.la \
     libumax1220u.la libumax_pp.la libv4l.la \
-    libxerox_mfp.la
+    libxerox_mfp.la libprimascan.la
 
 # Each stand alone backend thats possible to be built should be listed
 # here.  There are the libraries that are installed under $(libdir)/sane.
@@ -224,7 +224,7 @@
     libsane-tamarack.la libsane-test.la libsane-teco1.la \
     libsane-teco2.la libsane-teco3.la libsane-u12.la libsane-umax.la \
     libsane-umax1220u.la libsane-umax_pp.la libsane-v4l.la \
-    libsane-xerox_mfp.la
+    libsane-xerox_mfp.la libsane-primascan.la
 
 EXTRA_LTLIBRARIES = $(be_convenience_libs) $(be_dlopen_libs)
 
@@ -1064,6 +1064,14 @@
 libsane_xerox_mfp_la_LIBADD = $(COMMON_LIBS) libxerox_mfp.la ../sanei/sanei_init_debug.lo ../sanei/sanei_constrain_value.lo ../sanei/sanei_config.lo  sane_strstatus.lo ../sanei/sanei_usb.lo ../sanei/sanei_tcp.lo $(MATH_LIB) $(SOCKET_LIBS) $(USB_LIBS) $(RESMGR_LIBS)
 EXTRA_DIST += xerox_mfp.conf.in
 
+libprimascan_la_SOURCES = primascan.c primascan.h
+libprimascan_la_CPPFLAGS = $(AM_CPPFLAGS) -DBACKEND_NAME=primascan
+
+nodist_libsane_primascan_la_SOURCES = primascan-s.c
+libsane_primascan_la_CPPFLAGS = $(AM_CPPFLAGS) -DBACKEND_NAME=primascan
+libsane_primascan_la_LDFLAGS = $(DIST_SANELIBS_LDFLAGS)
+libsane_primascan_la_LIBADD = $(COMMON_LIBS) libprimascan.la ../sanei/sanei_init_debug.lo ../sanei/sanei_constrain_value.lo ../sanei/sanei_config.lo  sane_strstatus.lo ../sanei/sanei_usb.lo $(MATH_LIB) $(USB_LIBS)
+
 libdll_la_SOURCES =  dll.c
 BUILT_SOURCES = dll-preload.h
 CLEANFILES += dll-preload.h
