diff -p -up sane-backends-1.0.20/backend/rts8891.c.strformat sane-backends-1.0.20/backend/rts8891.c
--- sane-backends-1.0.20/backend/rts8891.c.strformat	2009-04-30 15:15:20.000000000 +0200
+++ sane-backends-1.0.20/backend/rts8891.c	2009-09-28 14:00:28.000000000 +0200
@@ -3751,7 +3751,7 @@ init_device (struct Rts8891_Device *dev)
       for (i = 0; i < dev->reg_count; i++)
 	sprintf (message + strlen (message), "0x%02x ", dev->regs[i]);
       sprintf (message + strlen (message), "\n");
-      DBG (DBG_io2, message);
+      DBG (DBG_io2, "%s", message);
     }
 
   /* initial sensor guess */
