diff -p -up sane-backends-1.0.23/acinclude.m4.pthread sane-backends-1.0.23/acinclude.m4
--- sane-backends-1.0.23/acinclude.m4.pthread	2011-11-21 21:17:48.000000000 +0100
+++ sane-backends-1.0.23/acinclude.m4	2012-12-19 14:58:51.673675096 +0100
@@ -245,7 +245,6 @@ AC_DEFUN([SANE_CHECK_PTHREAD],
       fi
     ])
 
-  if test $use_pthread = yes ; then
   AC_CHECK_HEADERS(pthread.h,
     [
        AC_CHECK_LIB(pthread, pthread_create, PTHREAD_LIBS="-lpthread")
@@ -256,16 +255,12 @@ AC_DEFUN([SANE_CHECK_PTHREAD],
 	,[ have_pthread=no; use_pthread=no ])
        LIBS="$save_LIBS"
     ],)
-  fi
  
   if test $use_pthread = yes ; then
     AC_DEFINE_UNQUOTED(USE_PTHREAD, "$use_pthread",
                    [Define if pthreads should be used instead of forked processes.])
-  else
-    dnl Reset library in case it was found but we are not going to use it.
-    PTHREAD_LIBS=""
   fi
-  if test "$have_pthread" = "yes" ; then
+  if test "$use_pthread" = "yes" ; then
     CPPFLAGS="${CPPFLAGS} -D_REENTRANT"
   fi
   AC_SUBST(PTHREAD_LIBS)
