--- sane-backends-1.0.21/doc/descriptions-external/brother2.desc.brother2list	2009-06-09 05:03:36.000000000 +0200
+++ sane-backends-1.0.21/doc/descriptions-external/brother2.desc	2010-05-18 10:39:47.000000000 +0200
@@ -56,6 +56,16 @@
 :interface "USB"
 :usbid "0x04f9" "0x0190"
 
+:model "DCP-130C"
+:status :untested
+:interface "USB"
+:usbid "0x04f9" "0x01a8"
+
+:model "DCP-195C"
+:status :untested
+:interface "USB"
+:usbid "0x04f9" "0x0222"
+
 :model "DCP-310CN"
 :status :untested
 :interface "USB network"
@@ -113,6 +123,12 @@
 :usbid "0x04f9" "0x01ab"
 :comment "Reportedly working."
 
+:model "MFC-260C"
+:status :untested
+:interface "USB"
+:usbid "0x04f9" "0x01d6"
+:comment "brscan2 needed."
+
 :model "MFC-3240C"
 :status :untested
 :interface "USB"
