From 2fc9de642fe0341918b1cebbe4079f645638aa7d Mon Sep 17 00:00:00 2001
From: Frederic Crozat <fcrozat@mandriva.com>
Date: Fri, 4 Sep 2009 15:06:37 +0200
Subject: [PATCH] fix scanner group

---
 tools/sane-desc.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/tools/sane-desc.c b/tools/sane-desc.c
index 600ab03..a72077c 100644
--- a/tools/sane-desc.c
+++ b/tools/sane-desc.c
@@ -58,7 +58,7 @@
 
 #define DEVMODE  "0664"
 #define DEVOWNER "root"
-#define DEVGROUP "scanner"
+#define DEVGROUP "usb"
 
 #ifndef PATH_MAX
 # define PATH_MAX 1024
-- 
1.6.4.2

