--- sane-backends-1.0.23/backend/Makefile.am.primascan	2012-07-25 02:02:14.000000000 +0200
+++ sane-backends-1.0.23/backend/Makefile.am	2012-08-28 11:08:14.753159014 +0200
@@ -191,7 +191,7 @@ be_convenience_libs = libabaton.la libag
     libtamarack.la libtest.la libteco1.la \
     libteco2.la libteco3.la libu12.la libumax.la \
     libumax1220u.la libumax_pp.la libv4l.la \
-    libxerox_mfp.la
+    libxerox_mfp.la libprimascan.la
 
 # Each stand alone backend thats possible to be built should be listed
 # here.  There are the libraries that are installed under $(libdir)/sane.
@@ -225,7 +225,7 @@ be_dlopen_libs = libsane-abaton.la libsa
     libsane-tamarack.la libsane-test.la libsane-teco1.la \
     libsane-teco2.la libsane-teco3.la libsane-u12.la libsane-umax.la \
     libsane-umax1220u.la libsane-umax_pp.la libsane-v4l.la \
-    libsane-xerox_mfp.la
+    libsane-xerox_mfp.la libsane-primascan.la
 
 EXTRA_LTLIBRARIES = $(be_convenience_libs) $(be_dlopen_libs)
 
@@ -1083,6 +1083,14 @@ libsane_xerox_mfp_la_LDFLAGS = $(DIST_SA
 libsane_xerox_mfp_la_LIBADD = $(COMMON_LIBS) libxerox_mfp.la ../sanei/sanei_init_debug.lo ../sanei/sanei_constrain_value.lo ../sanei/sanei_config.lo  sane_strstatus.lo ../sanei/sanei_usb.lo ../sanei/sanei_tcp.lo $(MATH_LIB) $(SOCKET_LIBS) $(USB_LIBS) $(RESMGR_LIBS)
 EXTRA_DIST += xerox_mfp.conf.in
 
+libprimascan_la_SOURCES = primascan.c primascan.h
+libprimascan_la_CPPFLAGS = $(AM_CPPFLAGS) -DBACKEND_NAME=primascan
+
+nodist_libsane_primascan_la_SOURCES = primascan-s.c
+libsane_primascan_la_CPPFLAGS = $(AM_CPPFLAGS) -DBACKEND_NAME=primascan
+libsane_primascan_la_LDFLAGS = $(DIST_SANELIBS_LDFLAGS)
+libsane_primascan_la_LIBADD = $(COMMON_LIBS) libprimascan.la ../sanei/sanei_init_debug.lo ../sanei/sanei_constrain_value.lo ../sanei/sanei_config.lo  sane_strstatus.lo ../sanei/sanei_usb.lo $(MATH_LIB) $(USB_LIBS)
+
 libdll_preload_la_SOURCES =  dll.c
 libdll_preload_la_CPPFLAGS = $(AM_CPPFLAGS) -DBACKEND_NAME=dll -DENABLE_PRELOAD
 libdll_la_SOURCES =  dll.c
--- sane-backends-1.0.23/configure.in.primascan	2012-08-19 22:31:09.000000000 +0200
+++ sane-backends-1.0.23/configure.in	2012-08-28 11:07:18.947949072 +0200
@@ -590,7 +590,7 @@ ALL_BACKENDS="abaton agfafocus apple art
         pixma plustek plustek_pp qcam ricoh rts8891 s9036 \
         sceptre sharp sm3600 sm3840 snapscan sp15c st400 \
         stv680 tamarack teco1 teco2 teco3 test u12 umax 
-        umax_pp umax1220u v4l xerox_mfp p5"
+        umax_pp umax1220u v4l xerox_mfp p5 primascan"
 
 # If user specifies backends manually then cause configure
 # to fail if its detected it can't be compiled.  If we
