diff -up sane-backends-1.0.30/backend/mustek_pp.c.open-macro sane-backends-1.0.30/backend/mustek_pp.c
--- sane-backends-1.0.30/backend/mustek_pp.c.open-macro	2020-06-16 00:48:19.577038357 +0200
+++ sane-backends-1.0.30/backend/mustek_pp.c	2020-06-16 00:49:32.308532702 +0200
@@ -1130,7 +1130,7 @@ sane_open (SANE_String_Const devicename,
 
 	}
 
-	if ((status = dev->func->open (dev->port, dev->caps, &fd)) != SANE_STATUS_GOOD) {
+	if ((status = (dev->func->open) (dev->port, dev->caps, &fd)) != SANE_STATUS_GOOD) {
 
 		DBG (1, "sane_open: could not open device (%s)\n",
 				sane_strstatus (status));
diff -up sane-backends-1.0.30/backend/plustek_pp.c.open-macro sane-backends-1.0.30/backend/plustek_pp.c
--- sane-backends-1.0.30/backend/plustek_pp.c.open-macro	2020-05-17 13:54:18.000000000 +0200
+++ sane-backends-1.0.30/backend/plustek_pp.c	2020-06-16 00:48:19.577038357 +0200
@@ -261,7 +261,7 @@ static int drvopen(	Plustek_Device *dev
 
     DBG( _DBG_INFO, "drvopen()\n" );
 
-	handle = dev->open((const char*)dev->name, (void *)dev );
+	handle = (dev->open)((const char*)dev->name, (void *)dev );
 
 	tsecs = 0;
 
diff -up sane-backends-1.0.30/sanei/sanei_scsi.c.open-macro sane-backends-1.0.30/sanei/sanei_scsi.c
--- sane-backends-1.0.30/sanei/sanei_scsi.c.open-macro	2020-05-17 13:54:18.000000000 +0200
+++ sane-backends-1.0.30/sanei/sanei_scsi.c	2020-06-16 00:48:19.577038357 +0200
@@ -5336,7 +5336,7 @@ sanei_scsi_find_devices (const char *fin
     (*plugInInterface)->Release (plugInInterface);
     IOObjectRelease (scsiDevice);
 
-    ioReturnValue = (*scsiDeviceInterface)->open (scsiDeviceInterface);
+    ioReturnValue = ((*scsiDeviceInterface)->open) (scsiDeviceInterface);
     if (ioReturnValue != kIOReturnSuccess)
       {
 	DBG (5, "Error opening SCSI interface (0x%08x)\n", ioReturnValue);
diff -up sane-backends-1.0.30/backend/pixma/pixma_common.c.omv~ sane-backends-1.0.30/backend/pixma/pixma_common.c
--- sane-backends-1.0.30/backend/pixma/pixma_common.c.omv~	2020-06-16 00:49:56.930038243 +0200
+++ sane-backends-1.0.30/backend/pixma/pixma_common.c	2020-06-16 00:50:06.657237924 +0200
@@ -780,7 +780,7 @@ pixma_open (unsigned devnr, pixma_t ** h
   strncpy (s->id, pixma_get_device_id (devnr), sizeof (s->id) - 1);
   s->ops = s->cfg->ops;
   s->scanning = 0;
-  error = s->ops->open (s);
+  error = (s->ops->open) (s);
   if (error < 0)
     goto rollback;
   error = pixma_deactivate (s->io);
