Index: doc/descriptions/epson.desc
===================================================================
--- ./doc/descriptions/epson.desc
+++ ./doc/descriptions/epson.desc	2010-05-03 09:47:04.000000000 +0200
@@ -172,7 +172,7 @@
 :status :complete
 
 :model "Expression 800"
-:interface "SCSI"
+:interface "SCSI" "EPSON" "Expression800" "processor
 :status :complete
 
 :model "Expression 1600"
Index: doc/descriptions/epson2.desc
===================================================================
--- ./doc/descriptions/epson2.desc
+++ ./doc/descriptions/epson2.desc	2010-05-03 09:48:31.000000000 +0200
@@ -239,7 +239,7 @@
 :comment    "overseas version of the GT-9500"
 
 :model      "Expression 800"		; command spec
-:interface  "SCSI"
+:interface  "SCSI" "EPSON" "Expression800" "processor
 :status     :complete
 :comment    "overseas version of the GT-9600"
 
